### ISSUE TITLE
body parameter isn't loaded when using parameter ref

### ISSUE THREAD

**Issue by msarmstr** [mission-liao/pyswagger#119]
Hi, I'm having an issue with using a ref for a body parameter.

```yaml
parameters:
  PetParam:
    in: body
    name: body
    required: true
    schema:
      $ref: '#/definitions/Pet'
paths: 
  /pet: 
    post:
      operationId: addPet
      parameters:
        - $ref: '#/parameters/PetParam'
```
When a request is made with the `addPet` operation above, the payload is empty.

Seems to be that `pyswagger.io.Request._prepare_body` isn't derefing the body parameter, so `getattr(parameter, 'in')` returns `None`.

By the way, this is a great tool. Thanks!

**Comment by mission-liao** [comment]
Hi, I think the problem in the sample spec you provide is because the Parameter Object is actually a "Reference Object" and pyswagger didn't deref it (i.e. use parameter.final instead of using it directly), I'll add a test case later and see if my guess is correct.

**Comment by mission-liao** [comment]
the fix should be included in latest build: v0.8.28.
